kube-client/src/config/file_config.rs

```diff
@@ -497,6 +497,33 @@ where
     });
 }
 
+fn read_path<P: AsRef<Path>>(path: P) -> io::Result<String> {
+    let bytes = fs::read(&path)?;
```

Comment:
on the face of it, this seems less safe of a change than catching an error from [`String::from_utf8`](https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8), but i've not seen this type of byte matching before. is this safe?

Reply:
we could do that considering that 99.9% of the cases files will be utf8, but in case if fails, we'll still need to do some byte matching to detect if it's utf16 LE or BE

as mentioned above, we could replace it with `encoding_rs::Encoding::for_bom` which would remove all the magic bytes

Reply:
hehe ok. i think encoding_rs is probably overkill for this with only this one special case. maybe in the future if the needs arise. as it stands this solution seems sufficiently self-contained and now tested.